### ISSUE TITLE
Gradle is already install, but build flutter will download gradle again 

### ISSUE THREAD

**Issue by hello-soon** [flutter/flutter#84868]
Gradle is already install, and set in the path
but, went I build flutter module, flutter will download gradle again 

# whereis gradle
gradle: /root/.sdkman/candidates/gradle/6.7/bin/gradle.bat /root/.sdkman/candidates/gradle/6.7/bin/gradle

# echo $PATH
/usr/local/git/bin:/usr/local/go/bin:/go/bin:/root/.sdkman/candidates/kotlin/current/bin:/root/.sdkman/candidates/gradle/current/bin:/root/.sdkman/candidates/maven/current/bin/:/usr/lib64/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/Flutter/bin:/usr/local/Flutter/bin/cache/dart-sdk/bin

# gradle --version

------------------------------------------------------------
Gradle 6.7
------------------------------------------------------------

Build time:   2020-10-14 16:13:12 UTC
Revision:     312ba9e0f4f8a02d01854d1ed743b79ed996dfd3

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          1.8.0_291 (Oracle Corporation 25.291-b10)
OS:           Linux 5.10.25-linuxkit amd64

# flutter doctor -v
[✓] Flutter (Channel unknown, 2.2.1, on Linux, locale zh_CN.UTF-8)
    • Flutter version 2.2.1 at /usr/local/Flutter
    • Framework revision 02c026b03c (3 周前), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /opt/app/android-sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /opt/app/android-sdk
    • Java binary at: /usr/java/jdk1.8.0_291-amd64/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_291-b10)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[!] Connected device
    ! No devices available

# env
ANDROID_HOME=/opt/app/android-sdk
FLUTTER_ROOT=
JAVA_HOME=/usr/java/jdk1.8.0_291-amd64
GRADLE_HOME=/root/.sdkman/candidates/gradle/current

----------------------

# flutter build aar --no-debug --no-profile --target-platform android-arm64 -v
[  +23 ms] executing: [/data/landun/workspace/trovo_flutter/.android/] /data/landun/workspace/trovo_flutter/.android/gradlew
-I=/usr/local/Flutter/packages/flutter_tools/gradle/aar_init_script.gradle -Pflutter-root=/usr/local/Flutter -Poutput-dir=/data/landun/workspace/trovo_flutter/build/host
-Pis-plugin=false -PbuildNumber=1.0 -Pverbose=true -Pdart-obfuscation=false -Ptrack-widget-creation=true -Ptree-shake-icons=true -Ptarget-platform=android-arm64
assembleAarRelease
[+6009 ms] Downloading https://services.gradle.org/distributions/gradle-6.7-all.zip

**Comment by sarn0ld** [comment]
As far as I can see, the path to the gradlew executable is hardcoded in a few files. I think the important ones could be:
https://github.com/flutter/flutter/blob/e467018d06c2658bd7502c329d66cfc121bcc4f1/dev/devicelab/lib/framework/apk_utils.dart
https://github.com/flutter/flutter/blob/c4a2a3e92d9b076cbcb7b76e7017410ddbc3bb5f/packages/flutter_tools/lib/src/android/gradle_utils.dart

It would be really nice if we could set the path to the gradle executable!

At the moment our CI Builds are wasting time by downloading the gradle wrapper on every build.

**Comment by zanderso** [comment]
Removing tool label to force triage by primary triage team.

**Comment by maheshmnj** [comment]
Hi @hello-soon, Thanks for filing the issue. Are there any known reproducible steps to reproduce the issue?

I tried running the app and it took 29.3 secs for gradle build and it did not redownload the gradle.

<details>
<summary>logs1</summary>

```
mahesh@Maheshs-MacBook-Air-M1 sample % flutter run
Multiple devices found:
Redmi K20 Pro (mobile)    • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
sdk gphone arm64 (mobile) • emulator-5554    • android-arm64  • Android 11 (API 30) (emulator)
macOS (desktop)           • macos            • darwin-arm64   • macOS 12.4 21F79 darwin-arm
Chrome (web)              • chrome           • web-javascript • Google Chrome 104.0.5112.79
[1]: Redmi K20 Pro (192.168.1.2:5555)
[2]: sdk gphone arm64 (emulator-5554)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Running "flutter pub get" in sample...                           1,980ms
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Note: /Users/mahesh/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-2.1.10/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Running Gradle task 'assembleDebug'...                             29.3s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                14.4s
Syncing files to device Redmi K20 Pro...                           116ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on Redmi K20 Pro is available at: http://127.0.0.1:52271/XFKxxpKWyuI=/
The Flutter DevTools debugger and profiler on Redmi K20 Pro is available at: http://127.0.0.1:9102?uri=http://127.0.0.1:52271/XFKxxpKWyuI=/
I/.example.sampl(10299): ProcessProfilingInfo new_methods=1003 is saved saved_to_disk=1 resolve_classes_delay=8000
``` 
</details> 

I then deleted the `android/.gradle` folder and after flutter clean it still took less time than previous build

<details>
<summary>logs2</summary>

```
mahesh@Maheshs-MacBook-Air-M1 sample % flutter run
Multiple devices found:
Redmi K20 Pro (mobile)    • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
sdk gphone arm64 (mobile) • emulator-5554    • android-arm64  • Android 11 (API 30) (emulator)
macOS (desktop)           • macos            • darwin-arm64   • macOS 12.4 21F79 darwin-arm
Chrome (web)              • chrome           • web-javascript • Google Chrome 104.0.5112.79
[1]: Redmi K20 Pro (192.168.1.2:5555)
[2]: sdk gphone arm64 (emulator-5554)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Running "flutter pub get" in sample...                           1,980ms
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Note: /Users/mahesh/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/google_maps_flutter-2.1.10/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Running Gradle task 'assembleDebug'...                             29.3s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                14.4s
Syncing files to device Redmi K20 Pro...                           116ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
mahesh@Maheshs-MacBook-Air-M1 sample % flutter clean
flutter pCleaning Xcode workspace...                                         3.5s
Cleaning Xcode workspace...                                      2,138ms
Deleting build...                                                  689ms
Deleting .dart_tool...                                              33ms
Deleting .packages...                                                0ms
Deleting Generated.xcconfig...                                       0ms
Deleting flutter_export_environment.sh...                            0ms
Deleting ephemeral...                                                0ms
Deleting ephemeral...                                                0ms
Deleting ephemeral...                                                0ms
Deleting .flutter-plugins-dependencies...                            0ms
Deleting .flutter-plugins...                                         0ms
mahesh@Maheshs-MacBook-Air-M1 sample % flutter pub get 
Running "flutter pub get" in sample...                           1,055ms
mahesh@Maheshs-MacBook-Air-M1 sample % flutter run    
Multiple devices found:
Redmi K20 Pro (mobile)    • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
sdk gphone arm64 (mobile) • emulator-5554    • android-arm64  • Android 11 (API 30) (emulator)
macOS (desktop)           • macos            • darwin-arm64   • macOS 12.4 21F79 darwin-arm
Chrome (web)              • chrome           • web-javascript • Google Chrome 104.0.5112.79
[1]: Redmi K20 Pro (192.168.1.2:5555)
[2]: sdk gphone arm64 (emulator-5554)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Running Gradle task 'assembleDebug'...                             19.1s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                16.8s
Syncing files to device Redmi K20 Pro...                           118ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on Redmi K20 Pro is available at: http://127.0.0.1:52823/ThzVAz4mhpo=/
The Flutter DevTools debugger and profiler on Redmi K20 Pro is available at:
http://127.0.0.1:9102?uri=http://127.0.0.1:52823/ThzVAz4mhpo=/

```
</details> 

Also, it seems you are using old version of flutter,  I would suggest upgrading to the latest stable release.

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by maheshmnj** [comment]
Similar to https://github.com/flutter/flutter/issues/99080
